### DISCUSSION TITLE
CCRotaryEncoder class absent in Platformio

### DISCUSSION THREAD

**Discussion by Atarity** [tttapa/Control-Surface#519] [Q&A] [answered]
I stuck with problem while the same code working fine with Arduino IDE and totally not with Platformio. I'm talking about [CCRotaryEncoder example](https://tttapa.github.io/Control-Surface-doc/Doxygen/d2/d1f/RotaryEncoder_8ino-example.html) with XIAO board.

**Control-Surface** installed with all dependencies (including **MIDIUSB** and **Encoder**) as [described in docs](https://tttapa.github.io/Control-Surface-doc/Doxygen/d8/da8/md_pages_Installation.html) for both Arduino IDE and Platformio.

While there is no problems with Arduino IDE, I've got below error with Platformio:
```
Compiling .pio/build/seeed_xiao/src/main.cpp.o
src/main.cpp:15:1: error: 'CCRotaryEncoder' does not name a type
 CCRotaryEncoder enc {
 ^~~~~~~~~~~~~~~
Compiling .pio/build/seeed_xiao/lib3f0/Seeed_Arduino_FreeRTOS/boards/SAMD51/FreeRTOSVariant.cpp.o
```

I can't see CCRotaryEncoder in IntelliSense while other classes are on place and working fine (I've checked CCButton — it works). Here it is my dependency tree from compilation log of Platformio:
```
Dependency Graph
|-- <Control Surface> 1.2.0-4
|   |-- <MIDIUSB> 1.0.5
|   |-- <SPI> 1.0
|   |   |-- <Adafruit Zero DMA Library> 1.0.4
|   |-- <Encoder> 1.4.1
|   |   |-- <Seeed Arduino FreeRTOS> 1.1.0
|   |-- <SoftwareSerial>
|   |-- <Seeed Arduino FreeRTOS> 1.1.0
```

I put the [complete sketch](https://gist.github.com/Atarity/2ebe60e1cd742efdcf9ac2797b75d7fb) for Paltformio to the Gist. Have no clue what I missing here. Do you guys and gals have any ideas?

**Comment by tttapa** [comment, marked as answer]
The example you linked to is from the master branch, whereas you have version 1.2.0 installed in PlatformIO. You can find the 1.2.0 documentation here: https://tttapa.github.io/Control-Surface-doc/1.2.0/Doxygen/d2/d1f/RotaryEncoder_8ino-example.html

As you can see, you have to include <Encoder.h> manually in the 1.x releases, whereas this is no longer needed in the newer master version.

You can find an overview of the differences between previous releases and the master branch at the bottom of the readme: https://github.com/tttapa/Control-Surface#change-log-and-updating

Alternatively, you could just install the master version in PlatformIO as well, since that's the one you used in the Arduino IDE.

**Reply by Atarity** [reply]
Thanks for timely answer!

I've already tried to include <Encoder.h> in `main.cpp` with 1.2.0-4 release — it gives nothing. But everything became fine just after I deleted 1.2.0-4 release from dependencies in `platformio.ini` and install library like 
```
pio lib install https://github.com/tttapa/Control-Surface.git#master
```
So thanks to pointing to diffs in documentation! It works now! 👍
